cloud_storage/google_cloud_storage/upload_file.py

```diff
@@ -0,0 +1,255 @@
+    """
+    Combine together the provided folder_name and file_name into one path
+    variable.
+    """
+    combined_name = os.path.normpath(
+        f'{folder_name}{"/" if folder_name else ""}{file_name}')
+    combined_name = os.path.normpath(combined_name)
+
+    return combined_name
+
+
+def determine_destination_full_path(
+        destination_folder_name,
+        destination_file_name,
+        source_full_path,
+        file_number=None):
+    """
+    Determine the final destination name of the file being downloaded.
+    """
+    destination_file_name = determine_destination_file_name(
+        destination_file_name=destination_file_name,
+        source_full_path=source_full_path,
+        file_number=file_number)
+    destination_full_path = combine_folder_and_file_name(
+        destination_folder_name, destination_file_name)
+    return destination_full_path
+
+
+def find_all_local_file_names(source_folder_name):
+    """
+    Returns a list of all files that exist in the current working directory,
+    filtered by source_folder_name if provided.
+    """
+    cwd = os.getcwd()
+    cwd_extension = os.path.normpath(f'{cwd}/{source_folder_name}/*')
+    file_names = glob.glob(cwd_extension)
+    return file_names
+
+
+def find_all_file_matches(file_names, file_name_re):
+    """
+    Return a list of all file_names that matched the regular expression.
+    """
+    matching_file_names = []
+    for file in file_names:
+        if re.search(file_name_re, file):
+            matching_file_names.append(file)
+
+    return matching_file_names
+
+
+def upload_google_cloud_storage_file(
+        gclient,
+        bucket,
+        source_full_path,
+        destination_full_path):
+    """
+    Uploads a single file to Google Cloud Storage.
+    """
+    blob = bucket.blob(destination_full_path)
+    blob.upload_from_filename(source_full_path)
+
+    print(f'{source_full_path} successfully uploaded to ' \
+            f'{bucket.name}/{destination_full_path}')
+
+
+def get_gclient():
+    """
+    Attempts to create the Google Cloud Storage Client with the associated
+    environment variables
+    """
+    try:
+        gclient = storage.Client()
+    except Exception as e:
+        print(f'Error accessing Google Cloud Storage with service account ' \
+                f'{args.gcp_application_credentials}')
+        sys.exit()
+
+
+def get_bucket(*,
+        gclient,
+        force_create,
+        bucket_name):
+    """
+    Fetches and returns the bucket from Google Cloud Storage and creates it
+    if force_create is set to True
+    """
+    if force_create:
+        try:
+            print(f'Creating bucket {bucket_name}')
+            gclient.create_bucket(bucket_name)
+        except Conflict as e:
+            print(f'Bucket {bucket_name} already exists')
+
+    try:
+        bucket = gclient.get_bucket(bucket_name)
+    except NotFound as e:
+        print(f'Bucket {bucket_name} does not exist\n {e}')
+        sys.exit()
```

Comment:
If we have to use sys.exit() because the core functionality was not completed (in this case uploading the file) we should return an exit code >=1 to indicate a failure.